### PR TITLE
Update dark theme and add supplemental dark to match DSM

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-1876.

#### Changes proposed in this Pull Request:
- Update dark theme and add supplemental dark to match DSM

### CODE REVIEW

**Comment by daileytj** [comment]
Posting this in 2 PR's for posterity, but let's leave all the comments in the react-native-showcase repo PR since the application is all demonstrated there.

We will still need to write up documentation for how to use the supplemental theme for dark theme as demonstrated in this project. I added a blueDarkSupplement theme to work around the limitations of react native paper's theming (@joebochill this was a great idea btw). This means that there are two dark themes. One will be provided globally using the PaperProvider and the supplemental one can be passed into all the proposed react native paper components via the theme prop.

The react native paper components that will need this applied for dark theme are:
- AppBar
- Bottom Navigation
- Button (only when using 'contained' variant)
- FAB
- TextInput
- ProgressBar
- SnackBar

We will need to update the RNCL as well (planning to do this as part of this story but would like themes and approach verified)

The following components may need updated to use this 2 part theming out of the box:
- Header (should be able to just add this to the internal appbar)
- List Item Tag (this isn't completely necessary since users can just pass in their own colors and the main dark primary color works just fine; If we want to default to our supplemental theme the way I'm using it in the showcase then we should update this.)
- Drawer
- InfoListItem (this is another optional one but I updated the background to just be theme.color.surface in all instances in the showcase. We could simplify this and use the theme by default in the component but I'm going to let someone else make this judgment call.)

Chips are a mess, I did a bunch of custom styles to get this working. I'm not seeing an easy way to do this with the theme(s). The source code for the chip component looks like they take in the theme.color.surface and modify it but it's not working as expected. This could be an opportunity to contribute to RNP or we could make a design pattern around the use of chips and suggest our custom styling, the former seems like the better approach at this point.

**Comment by daileytj** [comment]
https://user-images.githubusercontent.com/13989985/110701231-e8891780-81be-11eb-8c52-edb99df9c091.mov

**Review by joebochill — COMMENTED** [review]
Seems reasonable...i might just rename to blueDarkAltTheme or something similar.

**Comment by daileytj** [comment]
The dark theme updates should be good to go, the showcase still needs some work.

File: react-native/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Added
 
 -   Initial Dark theme
+-   Initial Dark theme alt
```
Comment (joebochill) [inline react-native/CHANGELOG.md:8]:
```suggestion
-   Initial Dark themes (regular and alternate)
```

File: react-native/src/blueDarkTheme.ts [hunk 2]
```diff
@@ -9,23 +9,25 @@ This code is licensed under the BSD-3 license found in the LICENSE file in the r
 import { configureFonts, DefaultTheme } from 'react-native-paper';
 import { blue, red, darkBlack, black, lightBlue } from '@pxblue/colors';
 import { fontConfig } from './shared';
-import { Theme } from 'react-native-paper/lib/typescript/src/types';
+import * as color from 'color';
```
Comment (joebochill) [inline react-native/src/blueDarkTheme.ts:12]: This is the typical way we import the color package (doesn't make a difference functionally, but for style consistency we may want to swap).
```suggestion
import Color from 'color';
```

Comment (joebochill) [reply react-native/src/blueDarkTheme.ts:12]: the uppercase kinda sorta helps distinguish from `theme.colors`

**Review by joebochill — APPROVED** [review]

**Review by joebochill — APPROVED** [review]

**Comment by joebochill** [comment]
Are you going to include some information in the Readme about how to use the two different themes?

**Pull request merged**